### PR TITLE
[@types/nova-editor-node] workspace.activeTextEditor can be null

### PR DESCRIPTION
**Pull request by eablokker**
Nova extension docs have been updated to say that activeTextEditor can be null
https://docs.nova.app/api-reference/workspace/#activetexteditor

I don't believe any tests will be affected.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/nova-editor-node/index.d.ts
Comment (apexskier) [inline types/nova-editor-node/index.d.ts:1012]: Is this `undefined`, or `null`?

**Review by apexskier — COMMENTED** [review]
Like the bot says, you'll need to add tests for this to be able to merge without a maintainers approval (I'm just a contributor).

**Comment by eablokker** [comment]
> Like the bot says, you'll need to add tests for this to be able to merge without a maintainers approval (I'm just a contributor).

I don't know how to write a test for this, since there was no existing test for it or for workspace.path which is similar, and it is not a function.

Is there an existing example of how to test for this?

File: types/nova-editor-node/index.d.ts
Comment (rbuckton) [reply types/nova-editor-node/index.d.ts:1012]: Worst case scenario, I'd just suggest this just in case:

```suggestion
    readonly activeTextEditor: TextEditor | null | undefined;
```

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/nova-editor-node/index.d.ts
Comment (eablokker) [reply types/nova-editor-node/index.d.ts:1012]: Here's Logan from Panic's reply:

> This is a consequence of using Apple’s JavaScriptCore and its Objective-C bridge (as Nova’s extension runtime is implemented in Objective-C, and bridged into JavaScript). It treats null and undefined as effectively the same thing in a lot of cases.
>
> We attempt to ensure it’s one or the other in a lot of cases, but most of the time if it can be null, it can also be undefined.

So it seems anything that can be null could also be undefined. What do you all think we should do? There are probably a lot of other properties in here that can be null | undefined.

**Review by andrewbranch — APPROVED** [review]

**Comment by eablokker** [comment]
Ready to merge

**Pull request merged**
